### PR TITLE
Fix ampersand handling in bulk translations

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?232557

@biyeun / @emord

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_translations.py [hunk 1]
```diff
@@ -23,8 +23,8 @@ def test_escape_output_value(self):
             ('unfortunate <xml expression', '<value>unfortunate &lt;xml expression</value>'),
             (u'क्लिक', '<value>&#2325;&#2381;&#2354;&#2367;&#2325;</value>'),
             ('&#39', '<value>&amp;#39</value>'),
-            ('Here is a ref <output value="/data/no_media" /> with some trailing text and bad < xml.',
-             '<value>Here is a ref <output value="/data/no_media"/> with some trailing text and bad &lt; xml.</value>')
+            ('Here is a ref <output value="/data/no_media" /> with some "trailing" text & that\'s some bad < xml.',
+             '<value>Here is a ref <output value="/data/no_media"/> with some "trailing" text &amp; that\'s some bad &lt; xml.</value>')
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_translations.py]: E501 line too long (136 > 115 characters)

File: corehq/apps/app_manager/translations.py [hunk 2]
```diff
@@ -379,7 +379,7 @@ def expected_bulk_app_sheet_rows(app):
                                 if isinstance(part, ItextOutput):
                                     value += "<output value=\"" + part.ref + "\"/>"
                                 else:
-                                    value += mark_safe(force_text(part).replace('<', '&lt;').replace('>', '&gt;'))
+                                    value += mark_safe(force_text(part).replace('&', '&amp;').replace('<', '&lt;').replace('>', '&gt;'))
```
Comment (dimagimon) [inline corehq/apps/app_manager/translations.py]: E501 line too long (136 > 115 characters)

**Comment by czue** [comment]
tests failed
```
======================================================================
FAIL: test_escape_output_value (corehq.apps.app_manager.tests.test_translations.AppManagerTranslationsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_translations.py", line 31, in test_escape_output_value
    self.assertEqual(expected_output, etree.tostring(escape_output_value(input)))
AssertionError: '<value>Here is a ref <output value="/data/no_media"/> with some "trailing" text &amp; that\'s some bad &lt; xml.</value>' != '<value>Here is a ref &lt;output value="/data/no_media" /&gt; with some "trailing" text &amp; that\'s some bad &lt; xml.</value>'
----------------------------------------------------------------------
```

File: corehq/apps/app_manager/tests/test_translations.py [hunk 3]
```diff
@@ -23,8 +23,10 @@ def test_escape_output_value(self):
             ('unfortunate <xml expression', '<value>unfortunate &lt;xml expression</value>'),
             (u'क्लिक', '<value>&#2325;&#2381;&#2354;&#2367;&#2325;</value>'),
             ('&#39', '<value>&amp;#39</value>'),
-            ('Here is a ref <output value="/data/no_media" /> with some trailing text and bad < xml.',
-             '<value>Here is a ref <output value="/data/no_media"/> with some trailing text and bad &lt; xml.</value>')
+            ('question1 is <output value="/data/question1" vellum:value="#form/question1"/> !',
+             '<value>question1 is &lt;output value="/data/question1" vellum:value="#form/question1"/&gt; !</value>'),
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_translations.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/app_manager/tests/test_translations.py [hunk 4]
```diff
@@ -23,8 +23,10 @@ def test_escape_output_value(self):
...
+            ('Here is a ref <output value="/data/no_media"/> with some "trailing" text & that\'s some bad < xml.',
+             '<value>Here is a ref &lt;output value="/data/no_media"/&gt; with some "trailing" text &amp; that\'s some bad &lt; xml.</value>')
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_translations.py]: E501 line too long (142 > 115 characters)

**Comment by orangejenny** [comment]
Fixed tests by updated expected value, which I realize looks sketchy, but I did manually test repeatedly downloading & re-uploading forms that use a mix of output refs and special characters, and uploading a file that contains special characters (`&`) and HTML entities (`&amp;`).

**Comment by orangejenny** [comment]
Did some testing with easy references both on and off.

**Pull request merged**